metrics/sinks/hawkular/driver_historical.go

```diff
@@ -0,0 +1,433 @@
+	tags[core.LabelMetricSetType.Key] = core.MetricSetTypeSystemContainer
+	r, err := h.client.TagValues(tags) // I assume these are stored at the system tenant
+	if r != nil {
+		return r[core.LabelContainerName.Key], err
+	}
+	return nil, err
+}
+
+// GetNamespaces retrieves the list of namespaces in the cluster
+func (h *hawkularSink) GetNamespaces() ([]string, error) {
+	// Fetch tenants, if the labelToTenantId is set (and includes namespace label)
+	if h.isNamespaceTenant() {
+		tds, err := h.client.Tenants()
+		if err != nil {
+			return nil, err
+		}
+		ns := make([]string, 0, len(tds))
+		for _, td := range tds {
+			ns = append(ns, td.ID)
+		}
+		return ns, nil
+	}
+
+	return h.getLabelTagValues(core.LabelNamespaceName.Key)
+}
+
+// Internal functions
+
+func (h *hawkularSink) getLabelTagValues(labelKey string, o ...metrics.Modifier) ([]string, error) {
+	tags := make(map[string]string)
+	tags[labelKey] = "*"
+	r, err := h.client.TagValues(tags, o...)
+	if r != nil {
+		return r[labelKey], err
+	}
+	return nil, err
+}
+
+// isNamespaceTenant tries to detect if namespace labels are used as target tenant. This is in use on Openshift as default (and Hawkular images)
+// if this returns true, the namespace information should be used when sending a query to the Hawkular, and only cluster level statistics are
+// fetched from the default tenant
+func (h *hawkularSink) isNamespaceTenant() bool {
+	// This could be a bit problematic.. are we really storing the right info?
+	if h.labelTenant != "" && (h.labelTenant == core.LabelPodNamespace.Key || h.labelTenant == core.LabelPodNamespaceUID.Key) {
+		return true
+	}
+	return false
+}
+
+func (h *hawkularSink) bucketPointToAggregationValue(bp *metrics.Bucketpoint, mt *metrics.MetricType, aggregations []core.AggregationType, bucketSize time.Duration) (core.TimestampedAggregationValue, error) {
+	aggregationValues := core.TimestampedAggregationValue{
+		Timestamp:  bp.Start,
+		BucketSize: bucketSize,
+		AggregationValue: core.AggregationValue{
+			Count: &bp.Samples,
+		},
+	}
+
+	aggs := make(map[core.AggregationType]core.MetricValue, len(aggregations))
+
+	for _, a := range aggregations {
+		var val float64
+		switch a {
+		case core.AggregationTypeAverage:
+			val = bp.Avg
+		case core.AggregationTypeMaximum:
+			val = bp.Max
+		case core.AggregationTypeMinimum:
+			val = bp.Min
+		case core.AggregationTypeMedian:
+			val = bp.Median
+		case core.AggregationTypePercentile50:
+			val = bp.Median // Percentile 50 and Median are the same thing
+		case core.AggregationTypePercentile95:
+			for _, v := range bp.Percentiles {
+				if v.Quantile == 95.0 {
+					val = v.Value
+				}
+			}
+		case core.AggregationTypePercentile99:
+			for _, v := range bp.Percentiles {
+				if v.Quantile == 99.0 {
+					val = v.Value
+				}
+			}
+		}
+
+		mv := core.MetricValue{
+			MetricType: core.MetricGauge,
+			ValueType:  core.ValueFloat,
+			FloatValue: float32(val),
+		}
+		f, err := metrics.ConvertToFloat64(val)
```

Comment:
missed this on my first pass through.  What's the point of calling this function?  We've already established that `val` converts to a `float32` directly above, so why call this function?